jsettlers.mapcreator/src/jsettlers/mapcreator/main/tools/ToolTreeModel.java

```diff
@@ -55,7 +54,7 @@ public Object getChild(Object parent, int index) {
 	@Override
 	public int getChildCount(Object parent) {
 		if (parent instanceof ToolBox) {
-			return ((ToolBox) parent).getTools().length;
+			return ((ToolBox) parent).getToolLength();
```

Comment:
I did some changes there, this code here is correct.
The templates are loaded from XML. It's possible to load multiple XML and therefore is possible to load also customized templates. (this is not implemented yet, only one file is loaded at the moment)